### ISSUE TITLE
the JSF

### ISSUE THREAD

**Issue by boneskull** [nodejs/bootstrap#5]
My background: JSF project owner (Mocha) and Node collaborator. I don’t participate in high-level governance of Node.js or either foundation. 

Today (Oct 13) we had meeting(s) regarding the foundation merger at the Node.js collab summit. I didn’t know this meeting was scheduled, but took the opportunity to observe. 

I had some notes/opinions which may or may not be of consequence, and in no particular order:

- The “top-level” JSF projects appear arbitrarily chosen by the ED based on criteria known only to the ED. I don’t believe the current standing of JSF projects should have ultimate bearing on where they land in a merged foundation. 
- @hackygolucky mentioned the projects were offered a seat at the TAC or something? Not sure. Nobody from the JSF asked anyone from my project about involvement. My point is that we cannot assume the JSF projects are disinterested in this process. e.g., libuv may be disinterested in the workings of a foundation, but until somebody asks the JSF project leads, we won’t know. 
- Mocha has historically few consistent contributors and is unlikely to ever attract a significant influx. But it’s relevant to the ecosystem and is growing in adoption. It *wants to make use of foundation resources*. I wouldn’t call it a candidate for the nursing home...but it’s not going to bring in much interest, corporate or otherwise. What of projects like this?
- JSF Projects have their own money from donations. I imagine the status quo is fine here, but some projects may need help from a foundation *actually spending the money*. 
- Projects were strongly discouraged (or outright told no) from using a DCO. Please not this again. Write it in the bylaws that projects have a choice. 
- Abysmal communication from JSF was pretty much expected, so the bar is set low. That said, nobody announced to the JSF projects that the discussion held today at the Node collab summit was going to happen and that they should attend (the town-hall was, however). 
- Project mentorship was nonexistent. If a project fails, it should not be because the foundation failed to deliver on its promises. I’d argue for mentorship to be a core value, perhaps...
- There’s a lot of mistrust, frustration and disappointment (and apathy. and indifference) amongst JSF project owners. There may be happy ones, but I have not met any. Here, we will need to make extra effort to engage them. Show them we have something to offer. Their trust must be earned—not simply expected. 

* * *

I probably won’t be involved in the process going forward, but felt welcome to offer my input (thanks). I hope this can be helpful—especially for those unfamiliar with the JSF—and used to productive ends. This is also just the opinion of one dude; I don’t claim to represent anybody else.

**Comment by boneskull** [comment]
Correction: mentorship was attempted but failed for lack of resources, structure, etc.

**Comment by benjamingr** [comment]
Random and slightly offtopic question for me to hide later @boneskull - when/if people ask me where to get involved in Open Source is Mocha (through a private email between you and the interested party) a good place to send them or would that be overcommitment on your behalf?

Mocha is a fun project everyone knows and it seems a lot easier to get into than Node (no giant download + compile)

**Comment by boneskull** [comment]
Also: I saw a lot of effort today and felt very good about that, so thanks again to all involved.

**Comment by rubys** [comment]
> nobody announced to the JSF projects that the discussion held today at the Node collab summit was going to happen and that they should attend (the town-hall was, however).

As someone who is ignorant as to how the JSF is organized: what would be the best way to announce to the relevant people at the JSF that **this** discussion/repository exists?

**Comment by amiller-gh** [comment]
> I had some notes/opinions which may or may not be of consequence

Of course they're of consequence!

> The “top-level” JSF projects appear arbitrarily chosen by the ED based on criteria known only to the ED. I don’t believe the current standing of JSF projects should have ultimate bearing on where they land in a merged foundation.

This is something we get to play with. We get to decide what this first round of project-leveling criteria looks like. (if the RFC process for new governance features from #7 turns out to be liked, presumably through that process). [Prior art is fun to look at here](https://github.com/cncf/toc/blob/master/process/graduation_criteria.adoc).

> @hackygolucky mentioned the projects were offered a seat at the TAC or something? Not sure. Nobody from the JSF asked anyone from my project about involvement. My point is that we cannot assume the JSF projects are disinterested in this process. e.g., libuv may be disinterested in the workings of a foundation, but until somebody asks the JSF project leads, we won’t know.

Not sure where that impression came from, but we're going to define expectations / process more in this repo, so we're going to see it become more clear. Now that we're writing it down in a public, collaborative place, these proposals will become far more concrete. 

> Mocha has historically few consistent contributors and is unlikely to ever attract a significant influx. But it’s relevant to the ecosystem and is growing in adoption. It wants to make use of foundation resources. I wouldn’t call it a candidate for the nursing home...but it’s not going to bring in much interest, corporate or otherwise. What of projects like this?

Good question. I believe in the current proposal it would live under the "Incubator" or "Commons.js" – these still need solid definitions. I believe @MylesBorins has given the most thought into this.

> JSF Projects have their own money from donations. I imagine the status quo is fine here, but some projects may need help from a foundation actually spending the money.

Will defer to @hackygolucky on monetary things here. Again, I think theres flexibility to do what we want here to an extent.

> Projects were strongly discouraged (or outright told no) from using a DCO. Please not this again. Write it in the bylaws that projects have a choice.

Lets discuss this in a separate thread – I think its a good ask. Once #6 or #7 land we can open a PR to add language to the expectations doc and have the discussion on details there.

> Abysmal communication from JSF was pretty much expected, so the bar is set low. That said, nobody announced to the JSF projects that the discussion held today at the Node collab summit was going to happen and that they should attend (the town-hall was, however).

Yup, I'm very sad about that... I'd echo rubys' question from above, what is the best way to reach out to everybody? I want to get them engaged, catch them up on discussions started at collab summit, and get us working together in this temporary repo. 

> Project mentorship was nonexistent. If a project fails, it should not be because the foundation failed to deliver on its promises. I’d argue for mentorship to be a core value, perhaps...

Core value may be a little too high-level for this to live, but it falls solidly under the "Community Focused" core value (added to the `/FOUNDATION.md` file in #7... we're going to slowly collect all the context from the many many conversations in this repo). I could see a top-level mentorship support program being of interest. We need to think about what that would look like and how the Foundation / TLCs can support that in a concrete way.

> There’s a lot of mistrust, frustration and disappointment (and apathy. and indifference) amongst JSF project owners. There may be happy ones, but I have not met any. Here, we will need to make extra effort to engage them. Show them we have something to offer. Their trust must be earned—not simply expected.

I @hackygolucky and @MylesBorins and I are more than willing to do the work. I think many many more who have been engaging here are too 🙂 You seem to have good context on some of the communication dynamics here. How do you think this group should best move forward with that outreach?

(Side note, this issue has a lot in it 😅 Lets focus this specific conversation on how to pull in more engaged folk from JSF member projects in a healthy way, and spin out the other programs / governance specific questions into their own issues / PRs)

**Comment by amiller-gh** [comment]
> Also: I saw a lot of effort today and felt very good about that, so thanks again to all involved.

I'm so friggn glad to hear that – I'm really happy with how the governance session turned out too! I thought while planning it that this process going to give me an aneurism... but everyone was amazing in the brainstorming room yesterday and it just re-enforced yet again why I love this community 😁 

That being said, I'm going to shoot out a survey soon for feedback on how those types of sessions can be improved in the future (what worked well, what didn't work well, what you'd like to see next time, etc.) So keep an eye out for that 👍

**Comment by knolleary** [comment]
> what is the best way to reach out to everybody?

The only thing I know of, as the lead of JSF project, is a mailing list: jsf-project-contacts@googlegroups.com - I don't know who can access it outside of the project leads and JSF leadership.

I'm sorry to have missed the in-person meeting, I wasn't aware it was happening and I had already scheduled my flights.

**Comment by boneskull** [comment]
pinging @dmethvin on the mailing list issue

**Comment by boneskull** [comment]
@rubys @amiller-gh 	

> As someone who is ignorant as to how the JSF is organized: what would be the best way to announce to the relevant people at the JSF that this discussion/repository exists?

The mailing list [Nick noted](https://github.com/nodejs/bootstrap/issues/5#issuecomment-430006195) is the only way I know how.  Traffic is very light, so hard to tell if people even bother to read it.

**Comment by boneskull** [comment]
@amiller-gh 	

> Not sure where that impression came from, but we're going to define expectations / process more in this repo, so we're going to see it become more clear. Now that we're writing it down in a public, collaborative place, these proposals will become far more concrete.

This came from @hackygolucky's comment during the meeting which made it seem like she had been told all JSF projects were disinterested in governance (unless I completely misunderstood).

> Good question. I believe in the current proposal it would live under the "Incubator" or "Commons.js" – these still need solid definitions. I believe @MylesBorins has given the most thought into this.

I'd like to talk with @MylesBorins and whoever else further about this in another issue.

**Comment by boneskull** [comment]
@amiller-gh 	

It appears not all JSF project leads are even *on* that mailing list for some reason.  A better way might be to visit each project's homepage or GitHub, see who is maintaining it, and email them directly.

**Comment by dmethvin** [comment]
I've sent an email on our internal list of project contacts and hope they'll add their relevant input here.

One of the goals of the JSF is to let projects manage themselves. Some like RequireJS are pretty much done and don't need day-to-day mantenance. Others like ESLint and jQuery have organized project teams and regular meetings. 

The process to get representation on the TAC is described at https://github.com/JSFoundation/TAC. Some projects decide they don't want or need that and just do their own thing, which is fine. The JSF does not add any requirement of "stuff to do" for projects since that would be a negative for many projects.

**Comment by jasnell** [comment]
@boneskull ... thank you for opening this thread. It raises a number of significant red flags that I think need to be dealt with fairly quickly.

/cc'ing @andybs to see if he can help provide some context here.

> The “top-level” JSF projects appear arbitrarily chosen by the ED
> based on criteria known only to the ED. I don’t believe the current
> standing of JSF projects should have ultimate bearing on where they
> land in a merged foundation.

If this is true, this is something that we will need to work through fairly quickly. Has there been any kind of "ranking" system used by JSF for projects?

> @hackygolucky mentioned the projects were offered a seat at the
> TAC or something? Not sure. Nobody from the JSF asked anyone from
> my project about involvement. My point is that we cannot assume
> the JSF projects are disinterested in this process. e.g., libuv
> may be disinterested in the workings of a foundation, but until
> somebody asks the JSF project leads, we won’t know.

@boneskull ... I'm curious... I'm much communication about the proposed merger was shared with the individual JSF projects *before* the intent to merge was announced?

> Mocha has historically few consistent contributors and is
> unlikely to ever attract a significant influx. But it’s
> relevant to the ecosystem and is growing in adoption. It
> wants to make use of foundation resources. I wouldn’t call
> it a candidate for the nursing home...but it’s not going
> to bring in much interest, corporate or otherwise. What of
> projects like this?

According to the process so far, each project would really decide for itself how much involvement it wants to have. However, for questions regarding representation in any kind of Cross Project Council or Foundation Board, we really do need to figure this out. It's not going to be feasible for every project to have a seat at the Board, but there are going to be projects that will likely insist on it (e.g. Node.js).

> JSF Projects have their own money from donations. I imagine
> the status quo is fine here, but some projects may need help
> from a foundation actually spending the money.

This is definitely something that will need to be figured out. One thing that has been discussed is the possibility of donations or corporate membership fees being earmarked for specific projects. It's not clear what that means for projects that don't receive any earmarks... 

> Projects were strongly discouraged (or outright told no) from
> using a DCO. Please not this again. Write it in the bylaws
> that projects have a choice.

+1. The choice of a DCO or CLA or whatever should be *entirely* up to each individual project. It should be the Foundation's role to support whatever mechanism the projects choose. The Foundation should never impose a specific choice on the individual projects.

> Abysmal communication from JSF was pretty much expected, so
> the bar is set low. That said, nobody announced to the JSF
> projects that the discussion held today at the Node collab
> summit was going to happen and that they should attend (the
> town-hall was, however).

That's deeply concerning. Have there been any plans made or discussed for a similar type of discussion among JSF projects?

> Project mentorship was nonexistent. If a project fails, it
> should not be because the foundation failed to deliver on
> its promises. I’d argue for mentorship to be a core value, perhaps...

Agreed.

> There’s a lot of mistrust, frustration and disappointment
> (and apathy. and indifference) amongst JSF project owners.
> There may be happy ones, but I have not met any. Here, we
> will need to make extra effort to engage them. Show them we
> have something to offer. Their trust must be earned—not simply expected.

This is something that I've seen from the fringes for a while. The merger discussions happening now are not new and it has been obvious for a while that there has been a high degree of dysfunction within the JS Foundation's interactions with the various projects.

At this point, I think the main thing that I would like to do is hear from the various project maintainers themselves about what they need and want moving forward. We've heard from the Foundation Board's about why they think this is a good idea. I don't think we really need to hear from them any more on it. Let's focus on the needs of all the individual projects and work forward from there.

**Comment by knolleary** [comment]
@jasnell 

> I'm curious... I'm much communication about the proposed merger was shared with the individual JSF projects before the intent to merge was announced?

There was a single email on Wednesday 3rd October sharing the news ahead of the announcement the following day that gave the high-level details. That was the first I knew of the discussions that had clearly been going on for some time. It had the promise of sharing the press-release before publication, but that didn't arrive.

>>  My point is that we cannot assume the JSF projects are disinterested in this process

Absolutely - I'm interested in the process, but I don't know how to engage with it. I only found out this repo exists because @boneskull pinged me on slack with a link. I was at Node+JS last week, but it seems the more in-depth conversations happened during the Node collaborators part of the event - I had already left at that point as I'm not a collaborator on the Node project.

My main observation tonight is almost all of the activity in this repo is coming from the Node fdn. There's nothing wrong with that, but I think its setting an impression of the JSF being merged into the Node fdn under some new name, rather than a true peer-to-peer merge.

> At this point, I think the main thing that I would like to do is hear from the various project maintainers themselves about what they need and want moving forward. 

As a project maintainer (Node-RED), I've had very little back from the JSF. There have been a couple offers of infrastructure (eg servers to host websites) which we've not needed at the time. There have been a couple opportunities to speak at events.

We are listed as a project in the mentorship lifecycle. I wasn't aware (or perhaps had forgotten though lack of engagement) until re-reading the links @dmethvin shared to the [Project Lifecycle](https://github.com/JSFoundation/TAC/blob/master/Project-Lifecycle.md) docs that we could graduate out of this phase and earn the right for representation on the TAC (I think... I may be misreading it). Looking at the criteria, I see we're barely any closer to reaching the targets it sets out than we were on the day we joined the JSF two years ago. The mentoring simply hasn't happened for us - or at least, there has been no proactive mentoring to nudge us along in any particular direction.

So, what would I want from a merged foundation? To be honest, I don't know as I'm not entirely sure what sorts of things a Foundation could do for us.

I *do* know I don't want lots of onerous new processes - but from everything I've seen so far, I'm not too worried that would be an outcome.

I think some more active engagement with the projects to help further both their own and the foundation's aims would be useful. A simple monthly/quarterly/whateverly checkpoint.

Someone taking a bit more of a holistic view across the various member projects would also help identify opportunities for collaboration - something that can be hard to do when you're singularly focussed on your own project and have little awareness of what the other projects are or what they are up to. I know the JSF has the concept of workgroups, but I'm not aware of any being set up.

Right now, its hard to tell a story about what binds the various projects in the JSF other than they happen to be written in the same language. Without that narrative, they projects remain a somewhat random collection and opportunities to cross-pollinate feel limited.


>> There’s a lot of mistrust, frustration and disappointment (and apathy. and indifference)

I will admit to having felt each of those at different times. I realise a lot of what I've written above sounds negative - and, well, yeah, a lot of it is. But I'm ultimately optimistic. There are many people looking at this with the right intentions and I hope we can distill out and reinforce the positives a combined fdn can bring.

**Comment by christian-bromann** [comment]
@jasnell 

Hey, I am maintaining the WebdriverIO project and joined the JSF last year. I agree with @knolleary on all of his points. I also was at Node+JS interactive but missed the meetings on Saturday. I wasn't really disappointed at all however there wasn't any proactive communication between the JSF champion that was assigned to me and myself. When I met him (which happen twice on conferences) it was more or less a "All good? Yes, all good!" communication. So there hasn't really changed much before and after joining the JSF which I think kind of failed the expectation I initially had.

I will observe the things going on here and jump in whenever I can share something valuable. Thank you all for being involved!

**Comment by dmethvin** [comment]
Sorry but there are a lot of things to respond to here and I don't have the time at the moment. I'll just pick off a couple.

> Projects were strongly discouraged (or outright told no) from using a DCO. Please not this again. Write it in the bylaws that projects have a choice.

One of the JSF projects ([Jerryscript](https://github.com/jerryscript-project/jerryscript)) uses a DCO and I'm the mentor, so I'm not sure where the discouraged came from but it's certainly not a prohibition. 

> So, what would I want from a merged foundation? To be honest, I don't know as I'm not entirely sure what sorts of things a Foundation could do for us. ... I do know I don't want lots of onerous new processes - but from everything I've seen so far, I'm not too worried that would be an outcome.

This is both a pro and a con of any loosely coupled organization. Within a pretty large range, you can either do your own thing or work with other projects to coordinate efforts. It's also possible that projects just doing their own thing don't realize they *could* collaborate with others, and that communication means that the mentorship and other communication needs more work.

**Comment by boneskull** [comment]
@dmethvin The ED said the DCO was for projects already using one prior to joining the foundation. Said he would follow up with legal on options but  never heard back.

**Comment by hackygolucky** [comment]
As a heads up @boneskull, a few of us pressed for more folks from the JSF to be a part of the conversation and particularly those engaged. When told about the process of how projects joined the TAC, we were told there were a number of steps and that many projects chose not to get involved at that level. Regardless, we kept asking for how we could reach those in the projects who would be interested. I was not told a mailing list existed. So!  That certainly makes thing different now if we can email to that list because it's vital for project members to participate for this to succeed. Our respective teams were responsible for reaching out to their relative groups for input and notification that the meetings were taking place. It sounds like we found a big bug in this process.

 I don't think I ever said the project members were not interested in governance, I simply reiterated what I'd been told about whether they were on the TAC. As this is not my group, I should have deferred to someone else. I would have loved to have been told I was wrong during that session 😁 That would have been helpful! 

I appreciate the clarification because I want to make sure we aren't discouraging anyone, or even worse, disenfranchising anyone.

Looking forward to us continuing to reach out to folks who are interested in participating in this work.

**Comment by boneskull** [comment]
@hackygolucky I’m not sure if you were even wrong—just that not necessarily everyone who would want to participate was offered a seat at the table.

**Comment by ilyavolodin** [comment]
Prefix: As a new parent, I'm a bit sleep deprived and cranky, so please take that into consideration. Also, a lot of people perceive me as negative, that's not my intention, I'm just "straight to the point" and don't think too much about what other meaning people might put into my words.
> That was the first I knew of the discussions that had clearly been going on for some time.

As a representative of ESLint on JSF TAC, I did know about this process before-hand. There were at least 2 calls that I dialed in, however, back when those calls happened, things were still very much in-the-air and I didn't get a huge amount of information, other then the merger is being planned.

> @boneskull ... I'm curious... I'm much communication about the proposed merger was shared with the individual JSF projects before the intent to merge was announced?

While the question wasn't directed at me, from ESLint point of view, I know about this happening, but all project representatives were asked on the call not to spread that information around, even to other maintainers of the projects.

>The “top-level” JSF projects appear arbitrarily chosen by the ED based on criteria known only to the ED. I don’t believe the current standing of JSF projects should have ultimate bearing on where they land in a merged foundation.

That was the question I asked on one of the calls (and in the email thread), what are the criteria on selection process for top projects from JSF? The answer was that it wasn't clear at the time, and it will be something that will be thought through in the future. Honestly, I was not satisfied with that answer. While the decision was being made by the foundations, it would directly affect all of the projects. And buying into something without having some of the more critical parts of the equation didn't seem right to me. My biggest concern was, and still is, that NodeJS is so much bigger project then any of the projects under JSF, it will clearly take priority in most of the decisions, leaving smaller projects behind. ESLint, Webpack and JQuery are all rather larger projects, but they don't really compare to NodeJS in size of either code base, participants or penetration (well, maybe jQuery comes somewhat close).

> As a project maintainer (Node-RED), I've had very little back from the JSF. There have been a couple offers of infrastructure (eg servers to host websites) which we've not needed at the time. There have been a couple opportunities to speak at events.

(The following is my personal opinion, and might not represent opinions of other maintainers of ESLint). **I** treat foundations as an insurance policy, based on my interactions with them. They are the copyright holders for all of the ESLint projects. If, at some point, somebody decides to sue us for something (unlikely to happen, but you never know) they will be the ones responsible for dealing with that. The other part is that if all of the ESLint maintainers would quit, there's a small possibility that the project might not die, and will continue to exist (I think that was the primary reason why we joined jQuery Foundation back in a day). Other then that, JSF does provide us with two servers, and some minor technical support for them. However, before we got those servers from JSF (that is getting them from DO), DO gave us 3 servers directly for free. So it wasn't a huge benefit for us. They also maintain our domain name and DNS records, as well as CDN account. Again, not really huge benefits as CDN is done through Cloudflare which is pretty good about giving free accounts to Open Source projects. Overall, I'm also not clear what other benefits can the foundation provide, so I treat it more as an insurance policy. Something you have to pay for (by giving up copyright for the project that you maintain), but when the trouble comes - worth having.
To be honest, I don't think that's the intention of the foundation. And I think the though of being treated as an insurance company might disgust some of the people involved (and that's probably a good thing). But, again, I'm not sure what is the point, other then that.

Here's what I would like to see from the new foundation:

* Clear guidelines of what makes the project "top" and represented on Cross Project Council
* Clear explanations of benefits and responsibilities of being on the Council or on the Board
* Very thorough explanation of benefits of being a part of the Foundation. What are things that Foundation can help with (vague paragraphs about support for advertisement and education doesn't count, as I don't really understand advertisement of Open Source Projects, and haven't seen an example of education. "Working with community" is also confusing, since we, as a project, already do that every single day, and do it reasonably well - enough to be used by a large number of people).
* Clear channels of communications on who to talk to, to request specific support (we've been told that JSF can help us with things like hiring a professional designer for our site redesign, or hiring profession information architect, to re-organize our documentation, we just need to contact foundation about it. However, I still have no idea who is it that I'm supposed to contact and how. I could reach out to Kris Borchers about it, but I'm not sure that's something that executive director deals with).
* Having a clear path for companies and individuals to donate money directly to specific projects is, I think, a critical part. There will be large number of projects in the new foundation. If all of the money are routed through the foundation itself, it will go to the biggest players (mainly NodeJS), that's just the reality of life, and nothing can be done about that. However, we had an instance of a company reaching out to ESLint and wanted to support us directly with monthly monetary contributions. We, as a project, didn't feel right bypassing foundation and taking money directly (and honestly, we have limited need for recurring paycheck), so we directed them to the foundation. As far as I know, that's where all of the communication died and the company is not donating to either us, or JSF.
* Most importantly - explanation about what are the steps that are being taken to make sure that not only the largest projects have a voice.
* A lot more documentation and instructions. For example, we are still not clear what are the steps when we want to create a new project under ESLint organization. Is it automatically becomes part of JSF, how to decide which dependent repositories should be part of the JSF and which shouldn't, etc.

P.S. Sorry again if it sounds like a rant, I probably should've brought up some of this stuff earlier either privately to JSF Board/Directors or earlier in the process of the merger. It just feels wrong to tell people who spend their time on establishing something that I still don't really understand what is it that they are doing and why (and still be a part of that something anyways). But since this seems like a pretty open discussion, I figured that it's probably better to put it out in the open, so some lessons could be learned from it. Also, a lot of the above is my personal opinion. ESLint as a project is content with the Foundation, and has no plans to leave.

**Comment by jlipps** [comment]
Hi all, JSF project maintainer (of [Appium](https://github.com/appium/appium)) here. Just to say that there _was_ a process for Appium "graduating" from the mentorship program to become a top-level project. It involved an open application and TAC approval. We had to substantiate our graduation with an application proving health/growth in a number of dimensions. Given that we were a founding JSF project and the first of these (new) projects to graduate, there was even a [press release](https://js.foundation/uncategorized/2017/08/29/appium-is-the-first-project-to-graduate-from-js-foundation-mentorship-program) of sorts about it (more interesting is the [application itself](https://github.com/JSFoundation/TAC/blob/master/Projects/Appium.md)).

FWIW I felt supported by the JSF in the process, though we pretty much had our stuff in order and didn't need a lot of help. But I was glad for the opportunity to work towards the graduation milestones.

Regarding JSF communication and project assistance, I've always found the JSF to be very responsive to the needs that we've had (primarily legal and infrastructure--the JSF has even ponied up cold hard cash to run some services for us, unless they worked some partner magic to get it for free). Earlier this year we had our first user conference and the JSF was very supportive and involved in helping us put that on.

As a TAC member I also feel like I did get at least a few notices about the merger in advance, though for my unrelated personal reasons I didn't engage with the conversations, though I would have liked to. IMO the merger is a good thing as long people can all agree on a model that works for the projects. The JSF was never a huge day-to-day piece of working on Appium so I'm optimistic that the merged foundation can be similar--something that runs in the background and helps take care of the boring non-technical, non-project-community stuff so we can focus on building awesome code and communities that work well together. That's the vision, right?

Of course it sounds like some projects have had a different experience and that should be investigated/resolved.

**Comment by boneskull** [comment]
@jlipps @ilyavolodin thanks for weighing in. 

There’s plenty we _could_ investigate or resolve... but little we _should_. It’s tempting—I have many questions for both of you—but it’s beating a dead horse. It won’t help us do what we’ve been tasked with.

**Comment by benjamingr** [comment]
@ilyavolodin thanks for writing that and being thoughtful - raising your concerns is important and it did not come off as "cranky" or "aggressive" IMO. It came off as "concerned" which is good.

I just want to say that for me as a Node.js collaborator (all the below is just my personal perspective):

> My biggest concern was, and still is, that NodeJS is so much bigger project then any of the projects under JSF, it will clearly take priority in most of the decisions, leaving smaller projects behind. ESLint, Webpack and JQuery are all rather larger projects, but they don't really compare to NodeJS in size of either code base, participants or penetration (well, maybe jQuery comes somewhat close).

I think it's important for Node.js to stop being that "much bigger" and possibly scary project and for us to acknowledge we're all a part of the same ecosystem and learn from each other. I think the "Node.js is a big project" atmosphere is a great motivation _for_ Node.js joining the JavaScript foundation and getting a seat at the _same table_ as other projects.

That is, as a Node.js project member I feel strongly about Node.js winning by "losing" its "status" as this "bigger project" and just being "a useful project". I think that would be a huge win that would encourage contribution to Node.js, make it more open to feedback and make it more open to new collaborators.

We're all part of the same ecosystem and care a lot about similar problems and I feel strongly that _your_ concerns as a maintainer of an important (and useful) tool are not overheard by Node.js.

I absolutely agree with all your points on the foundation needing to be clear and open about what it provides and to provide support. I'd also like to add to that list:

 - A clear avenue for projects like ESLint to provide feedback to standards bodies.
 - A clear avenue for projects like ESLint to gather community feedback (with help from statisticians hopefully).
 - Training with regards to running open source projects, shielding from "internet drama" and things that can negatively impact your health and moderation help.

Your post does not come off as a rant, thank you for speaking up and I hope in the future you feel safe doing so here. This is an important discussion to have and

**Comment by dylans** [comment]
HI there, to give some context, I'm Dylan (original co-founder of the first foundation (Dojo) and first foundation project (Dojo)), and also represent Intern as well.  Through my efforts with Dojo prior to the merger, I helped recruited many of the current projects to the foundation (Appium, Dojo, Grunt, Intern, Jed, Lodash, RequireJS, webpack, etc.). My company SitePen is also a member of the foundation and I'm involved with the marketing committee. Given that I've been here pretty much since the beginning of JS foundations (2004), I could take about this all day, but I'll try to keep things relatively brief.

Modern [Dojo](https://dojo.io.) and [Intern](https://theintern.io/) have not needed much from the foundation over the past year as both projects are running smoothly with active contributions and many new releases. What we have asked for primarily has been in the form of marketing help, and the primary reason that has been beneficial is that I've put in the effort and sit on the marketing committee.

I was a bit surprised to see that well-established and actively developed projects like jQuery, Dojo, Intern, and lodash are thought of as at-large projects, mostly because I don't pay a lot of attention to process related details. The mentorship process was added with the JS foundation, but established projects don't really need mentorship as they're more likely to mentor other projects. These are not dead projects, and some of the more active foundation projects as well as webpack, webhint, and several others. Activity is not the only measure of a thriving project (e.g. Mocha), but I think it's pretty easy to look at the JS foundation and see there are some projects that are near the end of their life probably (not to pick on them, but RequireJS and Sizzle), others that are very active, and others that are stable yet thriving in terms of adoption and usage.

The biggest complaint I've made to Kris and Dave (and anyone who will listen :) ) over the years is that the foundation needs to regularly reach out to the project leads and ask how they can help. Because I'm also a member, I have a few regular calls where I make sure Dojo and Intern get what they need, but my understanding is taht other projects have not received the same.

There are numerous lists we've created over the years of things that the foundation can do to better help foundation projects, but the JSF has been constrained by resources since its merger.

My other complain is that the biggest departure from the days of the Dojo Foundation to the days of the JS Foundation was sort of a default change from saying yes unless there was a really strong reason to say no, to saying no unless there's a really strong reason to say yes. Some of this is resource constraints, some of it is a function of available time, and some of it is becoming more formal under the foundation (which to be fair Dojo Foundation was not particularly by the book).

In general I'm an optimist. I think the vision for the JS foundation and a combined foundation is really strong, and a combined foundation can achieve more. The fundamental takeaway though is that in general, open source project leads are busy and don't really care about committees, foundation processes, etc. as along as we're heard and have a voice and have a way to object when things seem like they're going down a completely bad path.

We care about getting help and support when we need it to help our projects thrive, to fill in the gaps where we might struggle, etc. For example, with Dojo and Intern, the biggest need we have right now is how do we get people to even look at the nice projects we have created. But each project has different needs and in order to create a thriving foundation, the foundation leadership needs to engage the projects and do what they can to help projects achieve their full potential. Some projects want representation with standards bodies, or to make sure standards they need get some priority. Others need avenues towards raising money to fund development which the foundation has somewhat wisely avoided (I say this for many reasons, but in general a non-profit foundation cannot be in the business of funding all OSS development, though it can help guide projects down paths to funding that can be successful).

Overall, I would say that merging the Dojo and jQuery foundations to form the JS Foundation was the right call, but ended up down a very different path than I had expected going in! With the new foundation I think it will be important to clearly articulate the role of the foundation to the projects and its members, and to come up with a structure that feels efficient and helpful and transparent wherever possible.

Creating a bigger foundation with the energy of the various JS and Node foundation projects, and hopefully with an easier path towards more funding and resources as a combined group in order to be able to achieve more together is a good thing. As I noted during the event last week (which I really enjoyed fwiw), pretty much every JS Foundation project leverages Node in a meaningful way, we highly respect the work and community efforts Node has achieved, we know that takes a lot of work and that Node has had its own struggles along the way.

The amazing projects in the JS Foundation are what make the foundation what it is, and working together to create something bigger and helps bring the JS community even closer together would be really great in my opinion. As long as the new foundation makes sure each project gets what it needs, I think it will be successful.

**Comment by jasnell** [comment]
The comments here are excellent. If there are more, please keep them coming.

What is obvious from all of this is that active communication has been a challenge and a large number of assumptions have been made. I think the most critical thing we can take from this is that we should not make any assumptions about the level of activity or involvement from any of the individual JSF projects, and we should not arbitrarily assign "top level" status to any JSF project. Rather, we need to survey each individual project to get *their* point of view on where they fall and what their needs are.

**Comment by MylesBorins** [comment]
Removing the agenda tag. Is there more to discuss in this issue or could it be closed?

**Comment by boneskull** [comment]
It can be closed.
